docs/spfx/web-parts/guidance/getting-started-with-top-actions.md

```diff
@@ -0,0 +1,170 @@
+
+> At the time of writing this article, Top Actions only supports rendering a drop down and button command.
+
+
+![Top Actions Example](../../../images/webpart-top-actions.png)
+
+## Getting started
+
+> [!TIP]
+> These instructions assume you know [how to create a hello world web part](../get-started/build-a-hello-world-web-part.md).
+
+### 1. Define your Top Action configurations
+
+In the example below we are defining the callback function that will be used to pull the configurations for our Top Action commands.
+
+> Note: `getTopActionsConfiguration` must be defined as public on your web part's class.
+
+```typescript
+import { ITopActions } from '@microsoft/sp-top-actions';
+
+public getTopActionsConfiguration(): ITopActions | undefined {
+  return {
+    topActions: [],
+    onExecute: (actionName: string, newValue: any) => {}
+  };
+}
+```
+
+### 3. Define your toolbar's user interface
+
+The `topActions` array is an ordered list of controls to render in the web part toolbar. In the example below we are defining one top action as a button interface.
+
+```typescript
+import { PropertyPaneFieldType } from '@microsoft/sp-property-pane';
+
+return {
+  topActions: [
+    {
+      targetProperty: 'reset',
+      properties: {
+        icon: 'Reset'
+      },
+      type: PropertyPaneFieldType.Button
+    }
+  ]
+  ...
+}
+```
+
+### 4. Execute the command when the user interacts
+The previous step demonstrated how to get a button to display in the web part's toolbar. Now we will perform an action when the user clicks the button. Note that `actionName` was defined as `targetProperty` in the last step and since this is a button we can ignore the `newValue` that comes in.
+
+```typescript
+return {
+  ...
+  onExecute: (actionName: string, newValue: any) => {
+    if (actionName === 'reset') {
+      // user defined logic to reset the web part
+      this.reset();
+    }
+  }
+}
+```
+> Common pitfalls when implementing the `onExecute` command, is not syncing the new state with the web part properties and/or not refreshing or re-rendering the web part.
+
+## Code Snippets
+### Button command
+The type interace for a button is similar to the property panel's button (`IPropertyPaneButtonProps`).
+
+```typescript
+import { ITopActions } from '@microsoft/sp-top-actions';
+import { PropertyPaneFieldType } from '@microsoft/sp-property-pane';
+...
+public getTopActionsConfiguration(): ITopActions | undefined {
+  return {
+    topActions: [
+      {
+        targetProperty: 'reset',
+        type: PropertyPaneFieldType.Button,
+        properties: {
+          icon: 'Reset'
+        }
+      }
+    ],
+    onExecute: (actionName: string, newValue: any) => {
+      if (actionName === 'reset') {
+        // user defined logic to reset the web part
+        this.reset();
+      }
+    }
+  };
+}
+```
+
+### Drop down command

```

Comment:
"drop down" shouldn't be 2 words, should be one word or hyphenated according to MSDOCS

Reply:
Updated